### PR TITLE
Add EC Virgo

### PR DESCRIPTION
**Pull request by Cipulot**
## Description

Addition of the EC Virgo, the EC variant of the Virgo.

Piggybacked some cleanups in this PR too.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cipulot/ec_virgo/config.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+// #define DEBUG_MATRIX_SCAN_RATE
+
+#define EECONFIG_KB_DATA_SIZE 226
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (waffle87) [inline keyboards/cipulot/ec_virgo/config.h:72]:
```suggestion
```

File: keyboards/cipulot/ec_virgo/info.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
+{
+    "manufacturer": "Cipulot",
+    "keyboard_name": "Virgo EC",
+    "maintainer": "Cipulot",
+    "bootloader": "stm32-dfu",
+    "build": {
+        "lto": true
+    },
```
Comment (waffle87) [inline keyboards/cipulot/ec_virgo/info.json:8]:
```suggestion
    },
    "qmk": {
        "locking": {
            "enabled": true,
            "resync": true
        }
    },
```

File: keyboards/cipulot/ec_virgo/rules.mk [hunk 3]
```diff
@@ -0,0 +1,4 @@
+CUSTOM_MATRIX = lite
+ANALOG_DRIVER_REQUIRED = yes
+SRC += keyboards/cipulot/common/matrix.c keyboards/cipulot/common/ec_board.c keyboards/cipulot/common/ec_switch_matrix.c
```
Comment (waffle87) [inline keyboards/cipulot/ec_virgo/rules.mk:3]:
```suggestion
ANALOG_DRIVER_REQUIRED = yes
VPATH += keyboards/cipulot/common
SRC += matrix.c ec_board.c ec_switch_matrix.c
```

File: keyboards/cipulot/ec_virgo/info.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    "diode_direction": "COL2ROW",
+    "eeprom": {
+        "wear_leveling": {
+            "backing_size": 4096
+        }
+    },
+    "features": {
+        "bootmagic": false,
```
Comment (waffle87) [inline keyboards/cipulot/ec_virgo/info.json:16]: This is preferred to be enabled, unless there's a reason not to.
```suggestion
        "bootmagic": true,
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/cipulot/ec_virgo/info.json (hunk 4 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_virgo/info.json:16]: Bootmagic breaks some EEPROM and scan behavior as discussed in a chat in the QMK developer discord. I even later bypass it with a dummy call to be sure that it isn't re-enabled by via.
This should remain disabled

File: keyboards/cipulot/ec_virgo/rules.mk (hunk 3 above)
Comment (Cipulot) [reply keyboards/cipulot/ec_virgo/rules.mk:3]: Testing this. Once verified I'll apply. yhis and the locking items patch es to the remaining PRs open

Comment (Cipulot) [reply keyboards/cipulot/ec_virgo/rules.mk]: @waffle87 I tested and verified the patch. To save on unnecessary cloning of PRs I also applied the change for this include and the locking stuff to the other boards that are already merged.

**Comment by Cipulot** [comment]
Small bump to see if this can be moved forward, along with the rest of the EC pack after the required changes have been applied 😃

**Comment by Cipulot** [comment]
@waffle87 sorry for the ping, but this PR and the other I have open still are about to enter full production and are on track to be then shipped out to the customer. 
If no other changes are needed can they be merged so I can move on with other PRs? Thanks

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**